tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -105,7 +115,119 @@ propagationLevel is not (DbmPropagationLevel.Service or DbmPropagationLevel.Full
             // modify the command to add the comment
             var commandText = command.CommandText ?? string.Empty;
             var propagationComment = StringBuilderCache.GetStringAndRelease(propagatorStringBuilder);
-            if (ShouldAppend(integrationId, commandText))
+
+            if (command.CommandType == CommandType.StoredProcedure && integrationId == IntegrationId.SqlClient)
+            {
+                /*
+                * For CommandType.StoredProcedure, we need to modify the command text to use an EXEC statement and swap it to a CommandType.Text.
+                * Without doing this we _cannot_ inject the comment as it will be taken as part of the stored procedure name.
+                *    in SqlCommand.cs we have: rpc.rpcName = this.CommandText; // just get the raw command text
+                * Injecting the comment in the CommandText would result in getting an exception that the StoredProcedure name wasn't found (and would look like an empty string).
+                *
+                * We CAN NOT safely do this though if the Command has any parameters that are not Input (e.g. Return, InputOutput, Output).
+                *
+                * What this function does is:
+                *   - Swap to EXEC statement (which is a text command executing a stored procedure).
+                *   - Build a Input parameter list for the EXEC statement, so that we can pass all the parameters to it.
+                * Some helpful links:
+                * reference RPC for stored procedure: https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L5548
+                * reference text SQL (with params): https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L4488
+                * reference text SQL (no params): https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L4437
+                */
+
+                // check to see if we have any Return/InputOutput/Output parameters
```

Comment:
Is it worth logging a warning here, so that customers know _why_ they aren't getting propagation for some stored procedures 🤔 If so, we _might_ need to do some kind of rate limiting around it, so I'm unsure...

Reply:
I copy/pasted the "rate limiting" done in this file here as well 
Probably could do with some cleanup